torch/_dynamo/variables/ctx_manager.py

```diff
@@ -147,6 +147,37 @@ def fn_name(self):
         return "set_grad_enabled"
 
 
+class TorchFunctionDisableVariable(ContextWrappingVariable):
+    """represents whether torch function overrides are enabled or not"""
+
+    _guards_singleton = {
+        Guard("", GuardSource.GLOBAL, GuardBuilder.TORCH_FUNCTION_STATE)
+    }
+
+    @staticmethod
+    def create(tx, **kwargs):
+        var = TorchFunctionDisableVariable(
+            target_values=[False],
+            initial_values=[torch._C._is_torch_function_enabled()],
+            **kwargs,
+        )
+        var._call_func(tx, [False])
+        return var
+
+    def __init__(self, target_values, initial_values=None, **kwargs):
+        super().__init__(
+            target_values=target_values, initial_values=initial_values, **kwargs
+        )
+        self.guards = self.guards | self._guards_singleton
```

Comment:
If I clone (VTs get cloned all over), will I get multiple `_guards_singleton`? guards is a set, but unclear that these would hash the same for different instances...

Reply:
They're the same because class vars are only initialized once